tempesta_fw/http.c

```diff
@@ -877,16 +887,19 @@ tfw_http_req_cache_cb(TfwHttpReq *req, TfwHttpResp *resp)
 		spin_unlock(&srv_conn->msg_qlock);
 		goto send_500;
 	}
+	mb();
```

Comment:
It guarantees that the flag is set exactly after finishing `tfw_connection_send()` which can be inlined. I didn't use atomic operation for the flag manipulations, so I used explicit memory barrier.